packages/rest/src/session/AbstractSession.ts

```diff
@@ -256,86 +235,107 @@ export abstract class AbstractSession {
     private buildSession(session: ISession): ISession {
         const populatedSession = session;
 
-
         // set protocol if not set
-        if (isNullOrUndefined(populatedSession.protocol)) {
+        if (populatedSession.protocol === undefined || populatedSession.protocol === null) {
```

Comment:
A shorter replacement for `isNullOrUndefined(x)` is `x == null` (only 2 equal signs, so effectively handles both cases of "is null or undefined"). Not sure if this is important as `x === null || x === undefined` should be functionally identical. I only mention it because the suggestion of `x == null` has made to me on other PRs.

Reply:
I used what I did because that is what NodeJS recommends. I admit that does not make it the right answer.

https://nodejs.org/api/util.html#util_util_isnullorundefined_object

Reply:
Interesting that the link you provided says that they cannot guarantee backwards compatibility and that it has been deprecated since `v4.0.0`.

Should we open an issue so that we change all of these in a future PR?

Reply:
https://github.com/zowe/imperative/issues/400